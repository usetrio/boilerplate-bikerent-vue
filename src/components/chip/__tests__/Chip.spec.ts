import { describe, expect, it } from 'vitest'

import { Chip } from '@/components/chip'
import { mount } from '@vue/test-utils'

describe('Chip', () => {
  it('renders correctly', () => {
    const wrapper = mount(Chip, { slots: { default: 'content' } })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('prop size (default)', () => {
    const wrapper = mount(Chip, { slots: { default: 'content' } })

    expect(wrapper.classes()).contain('chip--sm')
  })

  it('prop color (default)', () => {
    const wrapper = mount(Chip, { slots: { default: 'content' } })

    expect(wrapper.classes()).contain('chip--gray')
  })

  it('prop size', () => {
    const wrapper = mount(Chip, { slots: { default: 'content' }, props: { size: 'lg' } })

    expect(wrapper.classes()).contain('chip--lg')
  })

  it('prop color', () => {
    const wrapper = mount(Chip, { slots: { default: 'content' }, props: { color: 'primary' } })

    expect(wrapper.classes()).contain('chip--primary')
  })
})
