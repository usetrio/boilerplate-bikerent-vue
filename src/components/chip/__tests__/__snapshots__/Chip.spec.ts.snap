// Vitest Snapshot v1

exports[`Chip > renders correctly 1`] = `
"<div class=\\"chip chip--sm chip--gray\\" data-v-221b4c58=\\"\\">
  <div class=\\"chip__content\\" data-v-221b4c58=\\"\\">content</div>
</div>"
`;
